lib/elasticity/index_config.rb

```diff
@@ -1,5 +1,9 @@
 module Elasticity
   class IndexConfig
+    class SubclassError < StandardError; end
+
+    SUBCLASSES_NOT_AVAILABLE = "subclasses are not available in this version of Elasticsearch".freeze
```

Comment:
Minor thing but I think it's important to give a more informative error message. What you think of: `Document type support have been removed from Elasticsearch 6, therefore document type based inheritance has been disabled by Elasticity`